### PR TITLE
Downgrade to 0.8.3 

### PR DESCRIPTION
**Pull request by jaeboklee-modo**
* Downgrade all contract version from 0.8.4 to 0.8.3
* New feature in solidity 0.8.4, the custom error, causes an error in some libraries including Slither. For stability and applying libraries, all contracts are downgraded to 0.8.3 and custom errors are replaced to require states.

### CODE REVIEW

File: contracts/DToken.sol [hunk 1]
```diff
@@ -70,7 +69,8 @@ contract DToken is IDToken, Context {
   function transfer(address recipient, uint256 amount) public virtual override returns (bool) {
     recipient;
     amount;
-    revert TokenErrors.DTokenTransferNotAllowed();
+    require(true != true, 'DTokenTransferNotAllowed');
+    revert();
```
Comment (donguks) [inline contracts/DToken.sol:73]: I think, `revert()` is never called.

File: contracts/DToken.sol [hunk 2]
```diff
@@ -70,7 +69,8 @@ contract DToken is IDToken, Context {
   function transfer(address recipient, uint256 amount) public virtual override returns (bool) {
     recipient;
     amount;
-    revert TokenErrors.DTokenTransferNotAllowed();
+    require(true != true, 'DTokenTransferNotAllowed');
```
Comment (donguks) [inline contracts/DToken.sol:72]: How about using `require(false, '~~~')`?

File: contracts/logic/Validation.sol [hunk 3]
```diff
@@ -18,9 +17,9 @@ library Validation {
    * @param amount Deposit amount
    **/
   function validateDeposit(DataStruct.ReserveData storage reserve, uint256 amount) internal view {
-    if (amount == 0) revert MoneyPoolErrors.InvalidAmount(amount);
-    if (reserve.isPaused == true) revert MoneyPoolErrors.ReservePaused();
-    if (reserve.isActivated == false) revert MoneyPoolErrors.ReserveInactivated();
+    require(amount != 0, 'InvalidAmount');
+    require(reserve.isPaused != true, 'ReservePaused');
+    require(reserve.isActivated != false, 'ReserveInactivated');
```
Comment (donguks) [inline contracts/logic/Validation.sol:22]: `isPaused` and `isActivated` are boolean.
I think, `!reserve.isPaused` and `reserve.isActivated` are enough.

File: contracts/logic/Validation.sol [hunk 4]
```diff
@@ -37,14 +36,12 @@ library Validation {
...
     uint256 userLTokenBalance
   ) internal view {
-    if (amount == 0) revert MoneyPoolErrors.InvalidAmount(amount);
-    if (reserve.isPaused == true) revert MoneyPoolErrors.ReservePaused();
-    if (reserve.isActivated == false) revert MoneyPoolErrors.ReserveInactivated();
-    if (amount > userLTokenBalance)
-      revert MoneyPoolErrors.WithdrawInsufficientBalance(amount, userLTokenBalance);
+    require(amount != 0, 'InvalidAmount');
+    require(reserve.isPaused != true, 'ReservePaused');
+    require(reserve.isActivated != false, 'ReserveInactivated');
```
Comment (donguks) [inline contracts/logic/Validation.sol:41]: It's the same here. `isPaused` `isActivated`

**Review by donguks — APPROVED** [review]
Good!
Eventually, the downgrade allows us to use slither :)

File: contracts/DToken.sol (hunk 1 above)
Comment (jaeboklee-modo) [reply contracts/DToken.sol:73]: Although these ERC20 functions are not used, it was necessary to specify the return type to override the interface of ERC20. However, the solidity compiler indicates an error for these expressions. These revert state were for removing errors in the Solidity compiler. Instead, I wrote meaningless return values!

File: contracts/DToken.sol (hunk 2 above)
Comment (jaeboklee-modo) [reply contracts/DToken.sol:72]: Ok applied.

File: contracts/logic/Validation.sol (hunk 3 above)
Comment (jaeboklee-modo) [reply contracts/logic/Validation.sol:22]: Great !

**Comment by jaeboklee-modo** [comment]
Thanks for reviewing.

**Pull request merged**